crates/client/src/client/schema/coins.rs

```diff
@@ -132,6 +136,98 @@ pub struct CoinIdFragment {
     pub utxo_id: UtxoId,
 }
 
+#[derive(cynic::InputObject, Clone, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl")]
+pub struct ExcludeInput {
+    /// Utxos to exclude from the result.
+    utxos: Vec<UtxoId>,
+    /// Messages to exclude from the result.
+    messages: Vec<Nonce>,
+}
+
+impl ExcludeInput {
+    pub fn from_tuple(tuple: (Vec<&str>, Vec<&str>)) -> Result<Self, ConversionError> {
+        let utxos = tuple.0.into_iter().map(UtxoId::from_str).try_collect()?;
+        let messages = tuple.1.into_iter().map(Nonce::from_str).try_collect()?;
+
+        Ok(Self { utxos, messages })
+    }
+}
+
+#[derive(cynic::InputObject, Clone, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl")]
+pub struct SpendQueryElementInput {
+    /// asset ID of the coins
+    pub asset_id: AssetId,
+    /// address of the owner
+    pub amount: U64,
+    /// the maximum number of coins per asset from the owner to return.
+    pub max: Option<U64>,
+}
+
+#[derive(cynic::QueryFragment, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl")]
+pub struct MessageCoin {
+    pub amount: U64,
+    pub sender: Address,
+    pub recipient: Address,
+    pub nonce: Nonce,
+    pub da_height: U64,
+}
+
+#[derive(cynic::InlineFragments, Debug)]
+#[cynic(schema_path = "./assets/schema.sdl")]
+pub enum Coins {
```

Comment:
Maybe `CoinType` since `Coins` is plural and implies multiple coins?

```suggestion
pub enum CoinType {
```